services/src/main/java/org/keycloak/services/resources/admin/permissions/UserPermissionsV2.java

```diff
@@ -118,10 +118,11 @@ private boolean hasPermission(UserModel user, EvaluationContext context, String.
 
         if (resource == null) {
             // check if there is permission for "all-users". If so, load its resource and proceed with evaluation
-            if (policyStore.findByName(server, AdminPermissionsSchema.USERS_RESOURCE_TYPE) == null) {
+            resource = AdminPermissionsSchema.SCHEMA.getResourceTypeResource(session, server, AdminPermissionsSchema.USERS_RESOURCE_TYPE);
+
+            if (policyStore.findByResource(server, resource).isEmpty()) {
```

Comment:
This query might be slow depending on how many permissions you have for the resource type. For now, it is OK, I think, and we can evaluate it later once we run some benchmarks before the release.